src/cursor/abstract_cursor.ts

```diff
@@ -388,7 +388,7 @@ export abstract class AbstractCursor<
     });
   }
 
-  close(): void;
+  close(): Promise<void>;
```

Comment:
I noticed this in some of my tests when calling close.  If we want to make this fix separately, I can break it out into a different PR.